zkevm-circuits/src/evm_circuit/op_execution/swap.rs

```diff
@@ -0,0 +1,483 @@
+use super::super::{
+    BusMappingLookup, Case, Cell, Constraint, CoreStateInstance, ExecutionStep,
+    FixedLookup, Lookup, Word,
+};
+use super::{CaseAllocation, CaseConfig, OpExecutionState, OpGadget};
+use crate::util::{Expr, ToWord};
+use bus_mapping::evm::{GasCost, OpcodeId};
+use halo2::{arithmetic::FieldExt, circuit::Region, plonk::Error};
+use std::{array, convert::TryInto};
+
+#[derive(Clone, Debug)]
+struct SwapSuccessAllocation<F> {
+    case_selector: Cell<F>,
+    words: [Word<F>; 2],
+}
+
+#[derive(Clone, Debug)]
+pub struct SwapGadget<F> {
+    success: SwapSuccessAllocation<F>,
+    stack_underflow: Cell<F>,
+    // case selector
```

Comment:
nitpick: Do you intend to comment like the suggested?
```suggestion
    stack_underflow: Cell<F>,  // case selector
```

Reply:
OK with it & updated